src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -3021,15 +3021,7 @@ interface FormatString {
         String toString();
     }
 
-    private static class FixedString implements FormatString {
-        private final String s;
-        private final int start;
-        private final int end;
-        FixedString(String s, int start, int end) {
-            this.s = s;
-            this.start = start;
-            this.end = end;
-        }
+    private static record FixedString(String s, int start, int end) implements FormatString {
```

Comment:
```suggestion
    private record FixedString(String s, int start, int end) implements FormatString {
```
Records are always implicitly static.